qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java

```diff
@@ -74,8 +74,9 @@ public void deleteVitals() {
 		return findElements(VITALS_LIST);
 	}
 
-	public void goToVisitNote() {
-		findElement(VISIT_NOTE).click();
+	public VisitNotePage goToVisitNote() {
+		clickOnLast(VISIT_NOTE);
+		return new VisitNotePage(this);
```

Comment:
How is this change related to what you are fixing?

Reply:
@kdaud , @sherrif10 the reason I implemented that method in this page was to cater for this step. 

` @When("a user clicks visit note link from the patient dashboard")
	public void loadVisitNotePage() {
		visitsDashboardPage = dashboardPage.goToRecentVisits();
		visitNotePage = (VisitNotePage) dashboardPage.goToVisitNote().waitForPage();
	}  `

so that it could be written like this 

` @When("a user clicks visit note link from the patient dashboard")
	public void loadVisitNotePage() {
		visitsDashboardPage = dashboardPage.goToRecentVisits();
		visitNotePage = (VisitNotePage) visitsDashboardPage.goToVisitNote().waitForPage();
	}  `

why would the test go to visitsDashboardPage and there after call goToVisitNote() from the dashboardPage, why not call it from visitsDashboardPage? What do you guys think?

Reply:
> why would the test go to visitsDashboardPage and there after call goToVisitNote() from the dashboardPage

ClinicianFacingdashboardPage is a parent page to VisitNotePage so a child page can access the functionality from the parent page as long as the functionality exists on the parent page.

> why not call it from visitsDashboardPage?

Yes the functionality can also be invoked directly on the child page.

Reply:
@kdaud so what can I implement in this case?

Reply:
@Irenyak1 is the existing implementation failing? And is the change related to what the ticket is addressing?

Reply:
@kdaud the implementation is not failing but there's code redundancy in this step
`@When("a user clicks visit note link from the patient dashboard") 
public void loadVisitNotePage() { 
visitsDashboardPage = dashboardPage.goToRecentVisits(); 
visitNotePage = (VisitNotePage) dashboardPage.goToVisitNote().waitForPage(); 
}`

Reply:
> @kdaud the implementation is not failing but there's code redundancy in this step

@Irenyak1 why not log a ticket for it and resolve. Did you get a chance to look at  this? https://opensource.com/article/18/6/anatomy-perfect-pull-request